extensions/amp-sidebar/0.1/amp-sidebar.js

```diff
@@ -160,7 +177,9 @@ export class AmpSidebar extends AMP.BaseElement {
         }, ANIMATION_TIMEOUT);
       });
     });
-    this.getHistory_().push(this.close_.bind(this)).then(historyId => {
+    this.getHistory_().push(() => {
+      this.close_();
```

Comment:
So event processing loop can possibly interfere here. In this case, though, waiting for one event loop (e.g. `setTimeout(..., 0)`) should be enough. Can you try that in your experiments. We just need this info to make a better decision.

Generally, I'd say, we don't have here enough information to make the right calls. We need more research and understand the exact sequence of events much better. If we need more data from sandbox - we should find a way to get it. One important thing: our history supports queuing and returns promises for `push` and `pop`. In theory, these promises should be sufficient to address any risk conditions. For this purpose I'm a little skeptical that we need timers.